modules/standard/Random.chpl

```diff
@@ -203,14 +204,29 @@ module Random {
                         seed: int(64) = SeedGenerator.oddCurrentTime,
                         param parSafe: bool = true,
                         param algorithm = defaultRNG) {
+    compilerWarning("makeRandomStream is deprecated - " +
+                    "please use RandomStream.create instead");
```

Comment:
It's not obvious to me why we need to deprecate makeRandomStream. It's doing something slightly different, isn't it? The `param algorithm` pattern exists also for a bunch of other functions here, so it seems to me like we should keep it.

Reply:
My main motivation was because it was one of the factory functions that are listed in #14291.

I understand your point that this is not a typical function, but at the same time I think it should follow the "convention" we are trying to set somehow, so maybe it should be just renamed to `createRandomStream` and we should ditch the idea of having `RandomStream.create` because it is not much different than `new owned RandomStream` as you point out below?

Reply:
It seems to me, if we want to update for #14291, we should just rename the function to `createRandomStream`.

Reply:
Only loosely following the discussion and PR(s), but it didn't seem like #14291 had reached a consensus on when to use `type.create()` vs. `createType()` IMO.

>  so maybe it should be just renamed to createRandomStream and we should ditch the idea of having RandomStream.create because it is not much different than new owned RandomStream as you point out below?

If this is the case, maybe we should consider removing `createRandomStream` and have users call `new`. Does the factory function save much effort compared to `new owned RandomStream`?

Reply:
It allows one to easily parameterize the RNG used (with the algorithm argument)

Reply:
OK, and just so I understand, does this represent the interfaces being considered?

```chpl
RandomStream.create(algorithm=PCG);
// same as
PCGRandomStream.create();
// same as 
new owned PCGRandomStream();

//
// vs.
//

createRandomStream(algorithm=PCG);
// same as
new owned PCGRandomStream();
```

Reply:
I think we decided not to make `PCGRandomStream.create();`.

Reply:
@ben-albrecht We currently have the second version of those, but the alternative was slightly different than your 1st version. It is more like

```chapel
RandomStream.create();
// same as (if defaultRNG==RNG.PCG, which is the default)
PCGRandomStream.create();
// same as
new owned PCGRandomStream();
```

My point being `RandomStream` is only a type alias to the default type, and not a type itself.